Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm

```diff
@@ -134,6 +134,75 @@ - (void)_updateMouseTouches:(NSSet<UITouch *> *)touches;
 - (void)_hoverGestureRecognized:(UIHoverGestureRecognizer *)gestureRecognizer;
 @end
 
+#if HAVE(UI_POINTER_INTERACTION)
+
+@interface TestPointerRegionRequest : UIPointerRegionRequest
+- (instancetype)initWithLocation:(CGPoint)location;
+@end
+
+@implementation TestPointerRegionRequest {
+    CGPoint _requestLocation;
+}
+
+- (instancetype)initWithLocation:(CGPoint)location
+{
+    if (!(self = [super init]))
+        return nil;
+
+    _requestLocation = location;
+    return self;
+}
+
+- (CGPoint)location
+{
+    return _requestLocation;
+}
+
+@end
+
+namespace TestWebKitAPI {
+
+struct PointerInfo {
+    BOOL isDefault { NO };
+    CGRect regionRect { CGRectNull };
+};
+
+} // namespace TestWebKitAPI
+
+@interface TestWKWebView (PointerInteractionTests)
+- (TestWebKitAPI::PointerInfo)pointerInfoAtLocation:(CGPoint)location;
+@end
+
+@implementation TestWKWebView (PointerInteractionTests)
+
+- (UIPointerInteraction *)pointerInteraction
+{
+    for (id<UIInteraction> interaction in self.textInputContentView.interactions) {
+        if (auto result = dynamic_objc_cast<UIPointerInteraction>(interaction))
+            return result;
+    }
+    return nil;
+}
+
+- (TestWebKitAPI::PointerInfo)pointerInfoAtLocation:(CGPoint)location
+{
+    auto contentView = (UIView<UIPointerInteractionDelegate> *)[self textInputContentView];
+    auto interaction = [self pointerInteraction];
+    auto request = adoptNS([[TestPointerRegionRequest alloc] initWithLocation:location]);
+
+    RetainPtr defaultRegion = [UIPointerRegion regionWithRect:contentView.bounds identifier:nil];
+    [contentView pointerInteraction:interaction regionForRequest:request.get() defaultRegion:defaultRegion.get()];
+    [self waitForNextPresentationUpdate];
+
+    auto region = [contentView pointerInteraction:interaction regionForRequest:request.get() defaultRegion:defaultRegion.get()];
```

Comment:
Hmm.. why are we calling `-[WKContentViewInteraction _pointerInteraction:]` twice? And throwing away the first result?

Reply:
Ah, so the first call returns the last cached (stale) data, and asynchronously triggers a position information update; the second call (after ensuring a round trip to the web process and back) with the same request location should then yield the updated pointer region at that location.